src/utils.jl

```diff
@@ -54,3 +54,53 @@ function checkbalanced(x::AbstractVecOrMat, y::AbstractVecOrMat)
         throw(ArgumentError("source and target marginals are not balanced"))
     return nothing
 end
+
+struct FiniteDiscreteMeasure{X<:AbstractArray,P<:AbstractVector}
```

Comment:
The issue is that in such cases, it wouldn't allow for something like `FiniteDiscreteMeasure(rand(10,2), p)`, since a matrix is not an abstract vector.

Reply:
I see. My bad for commenting before reading your link. I guess the `ColVecs` and `RowVecs` indeed disambiguate things.